### ISSUE TITLE
Native queries for Mongo and visualisation

### ISSUE THREAD

**Issue by anki-code** [metabase/metabase#2732]
I've tried to visualize this data:
In metabase: `[ { "$group" : { "_id" : "$EntityCreatedAt", "count": { "$sum": 1 } } } ]`
```
> db.Users.aggregate( [ { $group : { _id : "$EntityCreatedAt", count: { $sum: 1 } } } ] )
{ "_id" : ISODate("2016-06-08T14:03:22.104Z"), "count" : 1 }
{ "_id" : ISODate("2016-06-08T15:27:28.003Z"), "count" : 1 }
```

But the line chart (and others) unavailable:
![image](https://cloud.githubusercontent.com/assets/1708680/15979764/b5444874-2f6e-11e6-9b28-53031fe91575.png)

How I can fix it?

**Comment by agilliland** [comment]
you can still click on those options in the list even if they are dimmed out.

**Comment by anki-code** [comment]
In this case I'll get `We couldn’t create a line chart based on your query.`

**Comment by anki-code** [comment]
My version of build: `v0.18.0-snapshot (d40f59f master) `

**Comment by anki-code** [comment]
I've tried to aggregate the data by days:
```
> db.Users.aggregate([   
         {   $project : { day : {$substr: ["$EntityCreatedAt", 0, 10] }}},       
         {   $group   : { _id : "$day",  number : { $sum : 1 }}},
         {   $sort    : { _id : 1 }}       
     ])
{ "_id" : "2015-09-30", "number" : 129 }
{ "_id" : "2015-10-01", "number" : 226 }
```
And got the same error for the line chart: `We couldn’t create a line chart based on your query.`

**Comment by anki-code** [comment]
@agilliland @camsaul 
Any thoughts?

**Comment by agilliland** [comment]
Hmmm, you'll probably need to check out the JS console then and see if there are any more clues there.  Also, it would be helpful to post a sample of the data that is coming back from the server when you execute this query.  You can use the network console in your browser and check out the results from the call to the `dataset` endpoint.

@tlrobinson any thoughts on why @anki-code can't get the charts to render for his result?

**Comment by olmaga** [comment]
has anybody managed to plot a line/area with custom queries yet?

**Comment by tlrobinson** [comment]
@camsaul It looks like the problem is the backend returns `base_type: "UnknownField", special_type: null` for every column in a Mongo native query. Is this fixable on the backend?

![screenshot 2016-06-24 12 23 57](https://cloud.githubusercontent.com/assets/18193/16348308/8feccf0e-3a06-11e6-8b40-a5860fc2ae55.png)

**Comment by salsakran** [comment]
And why's this not in any of the backend tests? =)

**Comment by camsaul** [comment]
@tlrobinson We're supposed to be scanning the results and adding type info. I'm looking into this right now, it looks like this got broken when things moved around as part of mbql->native

I'll take this on.

@salsakran good question

**Comment by camsaul** [comment]
Actually we did have a test for this added as part of MBQL->native it but it was testing that the results were `:UnknownField` 😢 . 

I have a fix for this PR imminent

**Comment by camsaul** [comment]
fixed by #2835

**Comment by tlrobinson** [comment]
@anki-code We just released [0.18.1](https://github.com/metabase/metabase/releases/tag/v0.18.1) which should fix this. Thanks for reporting it.
